src/queryanswering/src/QueryInterface.cpp

```diff
@@ -293,4 +294,20 @@ void QueryInterface::filterResults(std::vector<std::string> &resultDesignators,
   }
   outInfo("Matching Object Descriptions: " << filteredResponse.size());
 }
+
+bool QueryInterface::getAnnotatorInOutConstraints(std::vector<std::string> &annotators, JsonPrologInterface::AnnotatorDependencies &dependencies)
+{
+    bool success = false;
+    try
+    {
+      success = jsonPrologInterface->retrieveAnnotatorsInputOutput(annotators, dependencies);
+    }
+    catch(...)
+    {
+      outError("jsonPrologInterface query failed!");
+    }
+
+    return success;
+}
+
```

Comment:
consider keeping this in the JsonPrologInterface class;the QueryInterface only serves for handling the input queries to RoboSherlock;

Reply:
ok, i wil change it tomorrow